packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -291,7 +291,14 @@ - (instancetype)initWithPlayerItem:(AVPlayerItem *)item
                 [self getVideoCompositionWithTransform:self->_preferredTransform
                                              withAsset:asset
                                         withVideoTrack:videoTrack];
-            item.videoComposition = videoComposition;
+            // Invalid values of video composition will throws an exception
+            // (https://github.com/flutter/flutter/issues/151031).
+            // When there is a problem with the parameters of video composition, set nil.
+            @try {
+              item.videoComposition = videoComposition;
+            } @catch (NSException *exception) {
+              item.videoComposition = nil;
```

Comment:
Errors, especially if they are ultimately caused by input from plugin clients, should never be silently ignored.

Reply:
I agree, as i said earlier,  if we know that the parameter is abnormal, should we still let it be created successfully? If the creation process is to continue, there should be no other way than to set fake data and not set it at all. Or just return an error back to flutter?

Reply:
As above; we cannot answer this question without more information about the cause of the error.